int32 g_vehicle_num
